### DISCUSSION TITLE
Mod screen feedback

### DISCUSSION THREAD

**Discussion by Joppe27** [ppy/osu#18194] [Ideas]
I've noticed the following points after playing around with the mod overlay for a good bit. This is mostly design-related, so just stuff to keep in mind. I'm not expecting this to immediately be taken into account if at all.

- Use entire dimmed screen area for horizontal scrolling

Aiming the mouse at the bottom scroll bar is annoying because the area right below it scrolls song select vertically. The other option, which is scrolling on the mod category headers, takes a while to figure out because it is not intuitive. I also do not see why you would want to scroll song select while having the mod overlay open, which is why I would use the entire dimmed screen area to avoid confusion.

https://user-images.githubusercontent.com/56885706/167483826-2a9195be-9386-417a-8d96-892af2360666.mp4

- Increase dim level of overlapped area

If above were to be agreed with, increasing dim makes it more obvious that the entire screen area can be used for scrolling. I also think the overlay looks incredibly busy as it is right now because of the low dim level.

- Switch difficulty multiplier colors

A mod that halves your score should not be given a happy green color. I checked stable, and the multiplier is red in this situation.
![image](https://user-images.githubusercontent.com/56885706/167485458-e8fad8f5-fdaa-4b85-b7f3-8eb7effff59b.png)

- Consider moving the difficulty multiplier

I know mobile is not a focus, but the mod overlay would work really really well for small screens if the difficulty multiplier had a different spot, even with 2x UI scaling. Currently, the multiplier takes up too much mod space on small screens.
![osu_2022-05-09_21-19-16](https://user-images.githubusercontent.com/56885706/167485954-0a5c63d2-47ae-43a9-b713-b9caef1bea10.jpg)

**Comment by bdach** [comment]
> Use entire dimmed screen area for horizontal scrolling

This was sorta the case before https://github.com/ppy/osu/pull/18133. Maybe could consider a revert? Not sure.

> Increase dim level of overlapped area

This is known.

> Switch difficulty multiplier colors

Would it not be more confusing if a mod from the green category caused a red mod multiplier, and vice versa?

> Consider moving the difficulty multiplier

There are more useless things on that screenshot there, like the header and toolbar.

**Reply by Joehuu** [reply]
> Would it not be more confusing if a mod from the green category caused a red mod multiplier, and vice versa?

Well, spun out is from automation and lowers the multiplier. Another argument is using mods from both categories, with the result of the multiplier always below 1x (for now that is).

So for the multiplier, I also think we shouldn't associate the colors with the sections, but generally (e.g. red is associated with danger), like how the OP described.

**Reply by frenzibyte** [reply]
And, well, doesn't "danger" come with more difficult mods 😅. It would feel really weird showing a green box on a 1.5x multiplier. The best I could imagine this to work is make the multiplier transition from no-colour to red on both directions and call it a day.

**Reply by Joehuu** [reply]
Yes, that is the reason why the difficulty increase section is red. In the multiplier sense when it's below 1x, your score is in "danger" (i.e. the maximum score you can get is lowered).

**Reply by peppy** [reply]
Green for lower difficulty is correct IMO.

**Reply by Joehuu** [reply]
Would using carets/arrows resolve this? Here's a rough idea of what I think it should look like:

![Kapture 2022-05-09 at 23 20 08](https://user-images.githubusercontent.com/35318437/167561482-ce7340c0-b764-45cf-b3e3-c4ed2828aee2.gif)

**Reply by peppy** [reply]
The colours are still reversed. Did you mean to have them inverted? If not what I say still 100% stands.

**Reply by Joehuu** [reply]
No, it's usual that up arrows are green and down arrows are red.

Reddit:
<img width="333" alt="Screen Shot 2022-05-10 at 10 49 34 AM" src="https://user-images.githubusercontent.com/35318437/167691027-73b20959-bda5-45f4-8868-59a2361f0032.png">

Wikipedia:
<img width="289" alt="Screen Shot 2022-05-10 at 10 51 47 AM" src="https://user-images.githubusercontent.com/35318437/167691453-a734534c-8add-4f84-9111-8f3054c9ebc8.png">

**Reply by bdach** [reply]
In isolation yes. But alongside the mod panels wherein green = easy and red = hard, my brain honestly gets confused even with the arrows.

**Reply by bdach** [reply]
Also consider this: there will likely be a bpm box nearby very soon. As I have it in my mind, it will turn green when bpm lowers and red when bpm increases, which matches the above. Would it not be *turbo* confusing to have that *and* the mod panels *and* the multiplier panels that would work the *opposite* way?

On this screen I want to keep the invariant: easy = green, hard = red. Not up or down or increase or decrease in numerical value. It's the only way to keep all of this non-confusing in my head.

**Reply by Joehuu** [reply]
To put another perspective, if `AdvancedStats` is getting the same treatment, would an AR 9 beatmap changed to AR 0 with difficulty adjust be considered ~green~ (actually red) here? This difficulty setting would need special consideration as the extremes are harder than the middle values.

**Reply by peppy** [reply]
decreasing difficulty = green
increasing difficulty = red

**Comment by peppy** [comment]
@Joppe27 as an aside, you can flick/drag scroll horizontally, instead of using the mouse wheel.

**Reply by papertek** [reply]
![osu_2022-05-09_23-54-26](https://user-images.githubusercontent.com/72106503/167545977-789ac763-06f8-45a6-930c-7b77562b829b.jpg)
personally i think scrolling should work similar to this photo (i dunno if that was the case before or has been suggested before but this makes it a lot easier to scroll in the mod select)

**Reply by Walavouchey** [reply]
why would an area outside a scroll container cause it to scroll?

**Reply by Joppe27** [reply]
> why would an area outside a scroll container cause it to scroll?

Have you ever used a web browser? The entire purpose of the scroll wheel is being able to position your mouse cursor wherever you want while still maintaining the ability to scroll. If we would limit the working area of the scroll wheel to the scroll countainer, why have a scroll wheel in the first place?

**Reply by Theighlin** [reply]
> why would an area outside a scroll container cause it to scroll?

You know how a scroll container works, but from a casual user persective, it's natural to try to scroll above the container since scrolling on the mod panels themselves causes the horizontal scrolling to stop working. 

It might not be intuitive that you can't scroll above there because said container has no visual borders on the upper side and not everyone will scroll precisely on the scrollbar or figure out you can do it on the section headers.

Not to say i agree with that image though, scrolling on the header or the bottom part of the overlay makes no sense, maybe an outline or background of some sort to only the panels would make it less confusing, not sure.

**Reply by Walavouchey** [reply]
> Have you ever used a web browser? The entire purpose of the scroll wheel is being able to position your mouse cursor wherever you want while still maintaining the ability to scroll. If we would limit the working area of the scroll wheel to the scroll countainer, why have a scroll wheel in the first place?

A web browser has a big scroll bar to the right of the window, which indicates the scrollable area. The mod selection overlay has its scroll bar above the bottom buttons, so should scrolling at either side of the scroll bar cause only one side to scroll? If yes, then what's the point of having the scroll bar on an edge of the scroll container? Let's have it in the middle!!

The design here clearly denotes only the middle part as scrollable. I could maybe agree to extending it upwards a bit, but as Theighlin said having the header or bottom part make the middle part scroll makes no sense, since they're entirely different regions.

**Comment by CenTdemeern1** [comment]
After using the new mod screen for a while, The only thing I can say is that it's basically impossible for me to get used to.
My brain doesn't really like horizontal lists. I wish there was a way to turn it into a vertical one. Maybe I'll try making something and PRing it.

**Reply by peppy** [reply]
If anything, I'd recommend waiting until we get better skin layout support, then resurrecting the previous version of mod select, which used horizontal scroll. We could potentially provide this for users that prefer to use it in the future (although it would probably need quite some refactoring to ensure it's maintainable).
